src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs

```diff
@@ -208,7 +208,13 @@ private static void ValidateTimestampCms(SigningSpecifications spec, SignedCms t
 
         private static void ValidateTimestampResponse(byte[] nonce, byte[] messageHash, IRfc3161TimestampToken timestampToken)
         {
-            if (!nonce.SequenceEqual(timestampToken.TokenInfo.GetNonce()))
+            var tokenNonce = timestampToken.TokenInfo.GetNonce();
+            if (tokenNonce == null)
+            {
+                throw new TimestampException(NuGetLogCode.NU3026, Strings.TimestampFailureNonceMismatch);
+            }
+
+            if (!nonce.SequenceEqual(tokenNonce))
```

Comment:
``` suggestion

            if (tokenNonce == null || !nonce.SequenceEqual(timestampToken.TokenInfo.GetNonce()))
```

@omajid - Thank you for addressing the previous code review comment. I did spend some time going through the code to check if the `nonce` argument value can be `null` and got convinced that we don't need a `null` check because `nonce` is generated and passed to the constructor. If you can address above suggestion, then I will run CI on this change.

https://github.com/NuGet/NuGet.Client/blob/a03a5a8cdd0cec5981fe2c1da6a683a7fee76622/src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs#L90-L97

Reply:
@omajid - Can you please address above code review comment? Please let us know if you don't have enough time to contribute to this PR which is understandable. If that is the case, I will spend some-time to take this PR to finish line.

Reply:
Hey, @kartheekp-ms, does the updated PR look okay to you?

Reply:
@omajid - Thank you for your contribution. I'll merge this PR by end of this week or early next week just in case if other team members have any feedback.